pkg/host/store.go

```diff
@@ -165,12 +165,14 @@ func (s *Store[E]) Delete(_ context.Context, id string) error {
 		return s.delete(id)
 	}
 
-	now := time.Now()
-	obj.SetDeletedAt(&now)
-	obj.IncrementResourceVersion()
+	if obj.GetDeletedAt() == nil {
+		now := time.Now()
+		obj.SetDeletedAt(&now)
+		obj.IncrementResourceVersion()
 
-	if _, err := s.set(obj); err != nil {
-		return fmt.Errorf("failed to set object metadata: %w", err)
+		if _, err := s.set(obj); err != nil {
+			return fmt.Errorf("failed to set object metadata: %w", err)
+		}
 	}
 
 	s.enqueue(store.WatchEvent[E]{
```

Comment:
If a second delete request is sent, nothing happens in this code. Do we still need to trigger the delete event in this scenario?

Reply:
i don't know, so i kept previous logic. Minimally it will still trigger reconcile loop.

Reply:
It doesn't seem necessary to trigger the reconcile loop if there are no changes.

Reply:
it is the design problem, you don't retrigger reconcile loop only but you notified every goroutines which registered watch function. So i kept this logic. @lukasfrank you will know better. I agree with @so-sahu and it isn't needed probably but i wanted risk side effects with designed change.

Reply:
you have right, i checked Update function and when object doesn't contain changes, it doesn't send event. I will change it.

Reply:
fixed